csharp/src/Ice/Internal/BZip2.cs

```diff
@@ -1,505 +1,295 @@
-                    SafeNativeMethods.unixBZ2_1_bzlibVersion();
-                }
+                Console.Error.WriteLine($"warning: found {libNames} but entry point BZ2_bzlibVersion is missing.");
             }
-            _bzlibInstalled = true;
-        }
-        catch (EntryPointNotFoundException)
-        {
-            Console.Error.WriteLine("warning: found " + _bzlibName + " but entry point BZ2_bzlibVersion is missing.");
-        }
-        catch (TypeLoadException)
-        {
-            // Expected -- bzip2 lib not installed or not in PATH.
-        }
-        catch (BadImageFormatException ex)
-        {
-            string lib = _bzlibName;
-            if (!string.IsNullOrEmpty(ex.FileName))
+            catch (TypeLoadException)
             {
-                lib = ex.FileName; // Future-proof: we'll do the right thing if the FileName member is non-empty.
+                // Expected -- bzip2 lib not installed or not in PATH.
             }
-            Console.Error.Write("warning: " + lib + " could not be loaded (likely due to 32/64-bit mismatch).");
-            if (IntPtr.Size == 8)
+            catch (BadImageFormatException)
             {
-                Console.Error.Write(" Make sure the directory containing the 64-bit " + lib + " is in your PATH.");
-            }
-            Console.Error.WriteLine();
-        }
-
-        if (AssemblyUtil.isWindows)
-        {
-            _compressBuffer = (
-                byte[] dest,
-                ref int destLen,
-                byte[] source,
-                int sourceLen,
-                int blockSize100k,
-                int verbosity,
-                int workFactor) =>
-                {
-                    return SafeNativeMethods.windowsBZ2_bzBuffToBuffCompress(
-                        dest,
-                        ref destLen,
-                        source,
-                        sourceLen,
-                        blockSize100k,
-                        verbosity,
-                        workFactor);
-                };
-
-            _decompressBuffer = (
-                byte[] dest,
-                ref int destLen,
-                byte[] source,
-                int sourceLen,
-                int small,
-                int verbosity) =>
-                {
-                    return SafeNativeMethods.windowsBZ2_bzBuffToBuffDecompress(
-                        dest,
-                        ref destLen,
-                        source,
-                        sourceLen,
-                        small,
-                        verbosity);
-                };
-        }
-        else if (AssemblyUtil.isMacOS)
-        {
-            _compressBuffer = (
-                byte[] dest,
-                ref int destLen,
-                byte[] source,
-                int sourceLen,
-                int blockSize100k,
-                int verbosity,
-                int workFactor) =>
+                Console.Error.Write(
```

Comment:
It would be nicer to use a logger - the logger of the communicator that requested this initialization to compress or decompress a message.